src/VisualStudio/Core/Def/Implementation/Remote/RemoteHostClientServiceFactory.RemoteHostClientService.cs

```diff
@@ -196,10 +196,18 @@ private void SetRemoteHostBitness()
                 var checksums = AddGlobalAssets(cancellationToken);
 
                 // send over global asset
-                await client.TryRunRemoteAsync(
-                    WellKnownRemoteHostServices.RemoteHostService, _workspace.CurrentSolution,
-                    nameof(IRemoteHostService.SynchronizeGlobalAssetsAsync),
-                    (object)checksums, cancellationToken).ConfigureAwait(false);
+                try
+                {
+                    // this only return false if OOP doesn't exist. otherwise, this will throw
+                    // as usual. 
+                    await client.TryRunRemoteAsync(
+                        WellKnownRemoteHostServices.RemoteHostService, _workspace.CurrentSolution,
+                        nameof(IRemoteHostService.SynchronizeGlobalAssetsAsync), (object)checksums, cancellationToken).ConfigureAwait(false);
+                }
+                catch(RemoteHostClientExtensions.UnexpectedRemoteHostException)
```

Comment:
I would have expected that TryRunRemoteAsync would be catching UnexpectedRemoteHostException and converting that to a false. Otherwise, you're effectively throwing an OperationCancelledException that wasn't from your original cancellation token. Yes, it's a derived exception but you get to follow the rules of your base type.

Reply:
this is special case for OOP engine itself. other ones, doesn't need to catch this exception. and TryRunRemoteAsync is API outside of OOP engine uses, so it doesn't catch this exception and let VS to soft crash.